shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/MetaDataServiceImpl.java

```diff
@@ -93,23 +92,24 @@ public String createOrUpdate(final MetaDataDTO metaDataDTO) {
     
     @Override
     public int delete(final List<String> ids) {
-        List<MetaDataDO> metaDataDoList = metaDataMapper.selectByIdList(ids);
+        List<String> distinctIds = ids.stream().distinct().collect(Collectors.toList());
+        List<MetaDataDO> metaDataDoList = metaDataMapper.selectByIdList(distinctIds);
         if (CollectionUtils.isEmpty(metaDataDoList)) {
             return 0;
         }
-        int count;
-        List<MetaData> metaDataList = Optional.ofNullable(metaDataDoList).orElseGet(ArrayList::new)
+        List<MetaData> metaDataList = metaDataDoList
                 .stream().map(MetaDataTransfer.INSTANCE::mapToData).collect(Collectors.toList());
             
-        count = metaDataMapper.deleteByIdList(ids);
+        int count = metaDataMapper.deleteByIdList(distinctIds);
         eventPublisher.publishEvent(new DataChangedEvent(ConfigGroupEnum.META_DATA, DataEventTypeEnum.DELETE, metaDataList));
 
         return count;
     }
     
     @Override
     public String enabled(final List<String> ids, final Boolean enabled) {
-        List<MetaDataDO> metaDataDoList = Optional.ofNullable(metaDataMapper.selectByIdList(ids)).orElseGet(ArrayList::new);
+        List<String> distinctIds = ids.stream().distinct().collect(Collectors.toList());
```

Comment:
ids have distinct？

Reply:
In principle, there will be no duplication, but if it is requested directly through the interface, the parameters can be changed, resulting in repeated ids.